tools/build-plugin.js

```diff
@@ -202,16 +201,6 @@ export class SourceProcessorSet {
     // don't use iteration functions, so we can return (and start at #1)
     for (let i = 1; i < parts.length; i++) {
       const extension = parts.slice(i).join('.');
-      // We specially handle 'js' in the build tool, because you can't provide a
-      // plugin to handle 'js' files, because the plugin would need to be built
-      // with JavaScript itself!  Places that hardcode JS are tagged with
-      // #HardcodeJs.
```

Comment:
In fact, you *can* provide a plugin to handle `.js` files, that handler just can't be the only way of handling `.js` files, and in particular it can't handle `.js` files within the plugin package itself.